lib/src/charts/data/core/cartesian/cartesian_styling.dart

```diff
@@ -243,13 +248,31 @@ class CartesianAxisStyle extends Equatable {
   }
 }
 
-class AxisLabelBuilder {
-  final LabelBuilder? left;
-  final LabelBuilder? top;
-  final LabelBuilder? right;
-  final LabelBuilder? bottom;
+class AxisLabelConfig extends Equatable {
+  final LabelBuilder builder;
+  final bool constraintEdgeLabels;
+  final bool centerLabels;
+
+  const AxisLabelConfig({
+    required this.builder,
+    this.constraintEdgeLabels = false,
+    this.centerLabels = false,
+  }) : assert(
+          !(constraintEdgeLabels && centerLabels),
+          "Both centerLabels and constraintEdgeLabels cannot be true at the same time.",
+        );
```

Comment:
Not sure if this is a requirement. Will having both as true cause issues?

Reply:
@shounak-mulay There is a similar assert for `AxisLabelsRenderer`. If we didn't specify it here, then it will be thrown from there.

Reply:
Okay, got it.